neo/rawio/intanrawio.py

```diff
@@ -53,40 +56,90 @@ class IntanRawIO(BaseRawIO):
 
+        BaseRawIO.__init__(self)
         self.filename = filename
 
     def _source_name(self):
         return self.filename
 
     def _parse_header(self):
 
+        filename = Path(self.filename)
+        
+        if not filename.exists() or not filename.is_file():
+            raise FileNotFoundError(f"{filename} does not exist")
+
         if self.filename.endswith(".rhs"):
+            self.file_type = 'header-attached'
             self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhs(
                 self.filename
             )
+        # 3 possibilities for rhd files two use a plain header called 'info.rhd' and one format has the header
+        # attached to the actual binary file with data
         elif self.filename.endswith(".rhd"):
+            if filename.name == "info.rhd":
+                # first we have one-file-per-signal which is where one neo stream/file is saved as .dat files
+                if any((filename.parent / file).exists() for file in one_file_per_signal_filenames):
+                    self.file_type = 'one-file-per-signal'
+                    raw_file_paths_dict = create_one_file_per_signal_dict(filename.parent)
+                # then there is one-file-per-channel where each channel in a neo stream is in its own .dat file
+                else:
+                    self.file_type = 'one-file-per-channel'
+                    raw_file_paths_dict = create_one_file_per_channel_dict(filename.parent)
+            # finally the format with the header-attached to the binary file as one giant file
+            else:
+                self.file_type = 'header-attached'
+
             self._global_info, self._ordered_channels, data_dtype, header_size, self._block_size = read_rhd(
-                self.filename
+                self.filename, self.file_type
             )
 
         # memmap raw data with the complicated structured dtype
-        self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        # if header-attached there is one giant memory-map
+        if self.file_type == "header-attached":
+            self._raw_data = np.memmap(self.filename, dtype=data_dtype, mode="r", offset=header_size)
+        else:
+            self._raw_data = {}
+            for stream_index, (stream_index_key, sub_datatype) in enumerate(data_dtype.items()):
+                # for 'one-file-per-signal' we have one memory map / neo stream
+                if self.file_type == "one-file-per-signal":
+                    self._raw_data[stream_index] = np.memmap(
+                        raw_file_paths_dict[stream_index_key], dtype=sub_datatype, mode="r"
+                    )
+                # for one-file-per-channel we have one memory map / channel stored as a list / neo stream
+                else:
+                    self._raw_data[stream_index] = []
+                    for channel_index, datatype in enumerate(sub_datatype):
```

Comment:
This `datatype` is dangerously closed to `data_type`, maybe:
* data_type
* stream_data_type
* channel_data_type

Reply:
Done. Great idea to be clearer!